cmake/dependencies/catch.cmake

```diff
@@ -7,22 +7,33 @@ function(find_catch2)
         if(NOT IS_DIRECTORY ${openPMD_catch_src})
             message(FATAL_ERROR "Specified directory openPMD_catch_src='${openPMD_catch_src}' does not exist!")
         endif()
+    elseif(openPMD_catch_tar)
+        message(STATUS "Downloading Catch2 ...")
+        message(STATUS "Catch2 source: ${openPMD_catch_tar}")
     elseif(openPMD_USE_INTERNAL_CATCH)
         message(STATUS "Downloading Catch2 ...")
         message(STATUS "Catch2 repository: ${openPMD_catch_repo} (${openPMD_catch_branch})")
-        include(FetchContent)
     endif()
     if(TARGET Catch2::Catch2)
         # nothing to do, target already exists in the superbuild
-    elseif(openPMD_USE_INTERNAL_CATCH OR openPMD_catch_src)
+    elseif(openPMD_USE_INTERNAL_CATCH OR openPMD_catch_src OR openPMD_catch_tar)
         if(openPMD_catch_src)
             add_subdirectory(${openPMD_catch_src} _deps/localCatch2-build/)
         else()
-            FetchContent_Declare(fetchedCatch2
-                GIT_REPOSITORY ${openPMD_catch_repo}
-                GIT_TAG        ${openPMD_catch_branch}
-                BUILD_IN_SOURCE 0
-            )
+            include(FetchContent)
+            if(openPMD_catch_tar)
+                FetchContent_Declare(fetchedCatch2
+                    URL             ${openPMD_catch_tar}
+                    URL_HASH        ${openPMD_catch_tar_hash}
+                    BUILD_IN_SOURCE 0
```

Comment:
I would prefer `OFF` or `FALSE` for bool options, but it is a personal preference.

Reply:
I think that was not supported in the past for this specific option, will check again if that changed.

Reply:
You are right, this is a `BOOL` and I can make it more readable! :)
- [latest](https://cmake.org/cmake/help/latest/module/ExternalProject.html#build-step-options)
- [CMake 3.22](https://cmake.org/cmake/help/v3.22/module/ExternalProject.html)